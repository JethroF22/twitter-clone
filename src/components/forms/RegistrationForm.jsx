import React, { Component } from 'react';

class RegistrationForm extends Component {
  state = {
    username: '',
    password: '',
    confirmPassword: '',
    handle: '',
    email: ''
  };

  render() {
    return (
      <div>
        <h3>Create an account</h3>
        <form onSubmit={() => console.log('submitted')}>
          <label htmlFor="username">Username: </label>
          <input
            type="text"
            name="username"
            value={this.state.username}
            onChange={() => console.log('Username changed')}
          />
          <br />
          <label htmlFor="email">Email: </label>
          <input
            type="email"
            name="email"
            value={this.state.email}
            onChange={() => console.log('Email changed')}
          />
          <br />
          <label htmlFor="handle">Handle: </label>
          <input
            type="text"
            name="handle"
            value={this.state.handle}
            onChange={() => console.log('Handle changed')}
            placeholder="Example: @i_am_jerry"
          />
          <br />
          <label htmlFor="password">Password: </label>
          <input
            type="password"
            name="password"
            value={this.state.password}
            onChange={() => console.log('Password changed')}
          />
          <br />
          <label htmlFor="confirmPassword">Confirm Password: </label>
          <input
            type="password"
            name="confirmPassword"
            value={this.state.confirmPassword}
            onChange={() => console.log('Confirm Password changed')}
          />
          <br />
        </form>
      </div>
    );
  }
}

export default RegistrationForm;
