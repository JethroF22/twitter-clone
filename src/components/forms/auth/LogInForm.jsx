import React, { Component } from 'react';
import { connect } from 'react-redux';
import _ from 'lodash';
import validator from 'validator';

import { logIn } from '../../../actions/auth';
import {
  actionStatusMessages,
  errorMessages
} from '../../../../config/const.json';

const {
  SUCCESS_MESSAGE,
  FAILED_MESSAGE,
  IN_PROGRESS_MESSAGE
} = actionStatusMessages;
const { INVALID_CREDENTIALS } = errorMessages;

export class LogInForm extends Component {
  state = {
    email: '',
    password: '',
    errors: {}
  };

  onChange = e => {
    const { value, name } = e.target;
    this.setState(() => ({
      [name]: value,
      errors: {}
    }));
  };

  onSubmit = e => {
    e.preventDefault();

    const credentials = _.pick(this.state, ['email', 'password']);

    if (validator.isEmail(credentials.email)) {
      this.props.logIn(credentials).then(() => {
        if (this.props.actionStatus === SUCCESS_MESSAGE) {
          alert('successful');
        } else if (this.props.actionStatus === FAILED_MESSAGE) {
          this.setState(() => ({
            errors: {
              authenticationError: INVALID_CREDENTIALS
            }
          }));
        }
      });
    } else {
      this.setState(() => ({
        errors: {
          email: 'Invalid email address'
        }
      }));
    }
  };

  render() {
    return (
      <div>
        <h3>Log In</h3>
        <form onSubmit={this.onSubmit}>
          {this.state.errors.authenticationError && (
            <p>{this.state.errors.authenticationError}</p>
          )}
          {this.state.errors.email && <p>{this.state.errors.email}</p>}
          <label htmlFor="email">Email: </label>
          <input
            type="email"
            name="email"
            value={this.state.email}
            onChange={this.onChange}
          />
          <br />
          <label htmlFor="password">Password: </label>
          <input
            type="password"
            name="password"
            value={this.state.password}
            onChange={this.onChange}
          />
          <br />
          <button
            type="submit"
            disabled={this.props.actionStatus === IN_PROGRESS_MESSAGE}
          >
            Log In
          </button>
        </form>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  actionStatus: state.status.actionStatus,
  errorType: state.error.errorType,
  errorMessage: state.error.errorMessage
});

const mapDispatchToProps = dispatch => ({
  logIn: credentials => dispatch(logIn(credentials))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(LogInForm);
