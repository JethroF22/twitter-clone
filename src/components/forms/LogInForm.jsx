import React, { Component } from 'react';
import { connect } from 'react-redux';
import _ from 'lodash';

import { logIn } from '../../actions/auth';

class LogInForm extends Component {
  state = {
    email: '',
    password: '',
    errors: {}
  };

  render() {
    return (
      <div>
        <h3>Log In</h3>
        <form>
          {this.state.errors.authenticationError && (
            <p>{this.state.errors.authenticationError}</p>
          )}
          <label htmlFor="email">Email: </label>
          <input
            type="email"
            name="email"
            value={this.state.email}
            onChange={() => console.log('Email changed')}
          />
          <br />
          <label htmlFor="password">Password: </label>
          <input
            type="password"
            name="password"
            value={this.state.password}
            onChange={() => console.log('Password changed')}
          />
          <br />
          <button type="submit">Log In</button>
        </form>
      </div>
    );
  }
}

export default LogInForm;
