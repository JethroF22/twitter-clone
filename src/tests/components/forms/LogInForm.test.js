import React from 'react';
import { shallow } from 'enzyme';
import _ from 'lodash';

import { LogInForm } from '../../../components/forms/LogInForm';
import { users } from '../../seed/seed';

let wrapper, state, logInSpy, user;

describe('LogInForm', () => {
  beforeEach(() => {
    logInSpy = jest.fn().mockResolvedValueOnce(null);
    wrapper = shallow(<LogInForm logIn={logInSpy} />);
    user = users[1];
    state = {
      ...user,
      errors: {}
    };
    wrapper.setState(state);
  });

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should handle onChange', () => {
    wrapper.find('input[name="email"]').prop('onChange')({
      target: {
        value: user.email,
        name: 'email'
      }
    });
    expect(wrapper.state('email')).toBe(user.email);
  });

  test('should handle onSubmit', () => {
    wrapper.find('form').simulate('submit', { preventDefault: () => {} });
    expect(wrapper.state('errors')).toEqual({});
    expect(logInSpy).toHaveBeenLastCalledWith({
      email: user.email,
      password: user.password
    });
  });

  test('should display authentication error', () => {
    state = {
      ...user,
      email: 'notavalidemail',
      errors: {}
    };
    wrapper.setState(state);
    wrapper.find('form').simulate('submit', { preventDefault: () => {} });
    expect(wrapper.state('errors')).toEqual({
      email: 'Invalid email address'
    });
  });
});
