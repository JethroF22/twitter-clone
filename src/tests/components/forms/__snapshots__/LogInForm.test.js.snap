// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogInForm should render correctly 1`] = `
<div>
  <h3>
    Log In
  </h3>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="email"
    >
      Email: 
    </label>
    <input
      name="email"
      onChange={[Function]}
      type="email"
      value="nariroh@gmail.com"
    />
    <br />
    <label
      htmlFor="password"
    >
      Password: 
    </label>
    <input
      name="password"
      onChange={[Function]}
      type="password"
      value="password2"
    />
    <br />
    <button
      disabled={false}
      type="submit"
    >
      Log In
    </button>
  </form>
</div>
`;
